wave_optimization/scripts/plot_3d_data.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python2
+import csv
+from math import pow
+
+import numpy as np
+from mpl_toolkits.mplot3d import Axes3D
+import matplotlib.pylab as plt
+
+
+TEST_DATA_3D = "pts3d.dat"
+TEST_DATA_1 = "pts1.dat"
+TEST_DATA_2 = "pts2.dat"
+
+
+def load_2d_data(fp, skip_header=True):
+    csv_file = open(fp, 'r')
+    csv_reader = csv.reader(csv_file)
+    if skip_header:
+        next(csv_reader, None)
+
+    data = {
+        "x": [], "y": [],
+    }
+    for line in csv_reader:
+        data["x"].append(float(line[0]))
+        data["y"].append(float(line[1]))
+
+    return data
+
+
+def load_3d_data(fp, skip_header=True):
+    csv_file = open(fp, 'r')
+    csv_reader = csv.reader(csv_file)
+    if skip_header:
+        next(csv_reader, None)
+
+    data = {
+        "x": [], "y": [], "z": [],
+    }
+    for line in csv_reader:
+        data["x"].append(float(line[0]))
+        data["y"].append(float(line[1]))
+        data["z"].append(float(line[2]))
+
+    return data
+
+
+def rotation_matrix(q):
+    R_00 = 1.0 - 2.0 * pow(q[1], 2) - 2.0 * pow(q[2], 2)
+    R_01 = 2.0 * q[0] * q[1] + 2.0 * q[3] * q[2]
```

Comment:
Why is a rotation matrix defined in a plotting function.  Shouldn't this be standardized?